onnx/version_converter/adapters/broadcast_forward_compatibility.h

```diff
@@ -0,0 +1,66 @@
+// Adapter for broadcasting ops in default domain from version 6 to 7
+
+#pragma once
+
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+class BroadcastForwardCompatibility final : public Adapter {
+  public:
+    explicit BroadcastForwardCompatibility(const std::string& op_name, const OpSetID&
+      initial, const OpSetID& target): Adapter(op_name, initial, target) {}
+
+    void adapt_broadcast_forward_compatibility(std::shared_ptr<Graph> graph, Node* node)
+      const {
+      // Remove axis and broadcast attributes
+      // Assess whether axis requires reshaping
+      if (node->hasAttribute(kbroadcast)) {
+        const ArrayRef<Value*>& inputs = node->inputs();
+        ONNX_ASSERTM(inputs.size() == 2, "%s in opset version 6 can only broadcast"
+          " between 2 inputs", name().c_str());
+        ONNX_ASSERTM(inputs[0]->has_sizes(), "Shape of first input not available.");
+        std::vector<Dimension> A_sizes = inputs[0]->sizes();
+        ONNX_ASSERTM(inputs[1]->has_sizes(), "Shape of second input not available.");
+        std::vector<Dimension> B_sizes = inputs[1]->sizes();
+        // Also assert that broadcasting syntax are correct if axis is not present
+        if (node->hasAttribute(kaxis)) {
+          if (!onnx_opset7_broadcastable(node->i(kaxis), A_sizes, B_sizes)) {
```

Comment:
This if-else seems incorrect.

Reply:
How would you recommend naming this function?  It asserts that the op is opset 7 broadcastable, but returns whether or not it requires broadcasting.  Should I call it onnx_opset7_requires_broadcasting?